### PR TITLE
90 - Refactor and tidy up code

### PR DESCRIPTION
**Pull request by juliano60**
- Refactor ExampleClient
- Add color output support
- Remove unused packages
- Fix formatting issues

### CODE REVIEW

**Review by claudist — DISMISSED** [review]
I like the formatting update! looks good!

File: src/main/java/newbank/client/ExampleClient.java [hunk 1]
```diff
@@ -44,21 +54,44 @@ public void run() {
               String response = null;
               try {
                 response = bankServerIn.readLine();
+                gotReply = true;
```
Comment (michowski) [inline src/main/java/newbank/client/ExampleClient.java:57]: Jules, I know it's the most vague question ever. But how confident are you in merging it? :D Seems like a "major" change to the "core engine".

File: src/main/java/newbank/server/NewBankClientHandler.java [hunk 2]
```diff
@@ -75,7 +75,10 @@ public void run() {
     try {
       while (hasMore) {
         String request = in.readLine();
-        out.println(String.format("Received request [%s]", request));
+        if (NewBankServer.VERBOSE_MODE) {
+          // useful for debugging/testing, not meant for production
+          out.println(String.format("Received request [%s]", request));
```
Comment (michowski) [inline src/main/java/newbank/server/NewBankClientHandler.java:80]: Aren't we relying on those in tests?

File: src/main/java/newbank/client/TestClient.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      throws IOException, IllegalThreadStateException, ConfigurationException {
+    super(ip, port, reader);
+  }
+
+  protected void displayPrompt() {
+    // NOT NEEDED FOR TEST CLIENT
+  }
+  
+  protected void processResponse(final String response) {
+	  display.writeLine(response);
```
Comment (michowski) [inline src/main/java/newbank/client/TestClient.java]: The formatting seems inconsistent in this file.

File: src/main/java/newbank/client/ExampleClient.java (hunk 1 above)
Comment (juliano60) [reply src/main/java/newbank/client/ExampleClient.java:57]: No, not really. I would rely on our battery of tests :)

File: src/main/java/newbank/client/TestClient.java (hunk 3 above)
Comment (juliano60) [reply src/main/java/newbank/client/TestClient.java]: Are you sure? Let me double check...

File: src/main/java/newbank/server/NewBankClientHandler.java (hunk 2 above)
Comment (juliano60) [reply src/main/java/newbank/server/NewBankClientHandler.java:80]: Yes. The tests rely on those so it hasn't changed.

File: src/main/java/newbank/client/TestClient.java (hunk 3 above)
Comment (juliano60) [reply src/main/java/newbank/client/TestClient.java]: Good catch, re-applied the formatting to that file.

Kindest Regards,
Jules

File: src/main/java/newbank/client/ExampleClient.java (hunk 1 above)
Comment (michowski) [reply src/main/java/newbank/client/ExampleClient.java:57]: Sounds good!

**Review by michowski — APPROVED** [review]
Let's gooooooooo :rocket:

**Comment by juliano60** [comment]
Cheers guys!

Kindest Regards,
Jules

**Pull request merged**